### ISSUE TITLE
craft\fields\Assets Unexpected behavior in "afterElementSave"

### ISSUE THREAD

**Issue by Anubarak** [craftcms/cms#6269]
### TL/DR
pelase see my comment [here  ](https://github.com/craftcms/cms/commit/885912438eb9c8929152552524d632bb0a6b514f#diff-5e0143828c37246b1af8a8380056c5ac)

### Description
When saving an element with an assets field I discovered a strange behavior and I would like to turn it off if it's intended or being fixed if it's not intended.

When you save an element **all** related assets will **always** be moved to a different volume if they are not in a default determined by Craft in this function
https://github.com/craftcms/cms/blob/3.5/src/fields/Assets.php#L454

```PHP
$getTargetFolderId = function() use ($element, $isDraftOrRevision): int {
    static $targetFolderId;
    return $targetFolderId = $targetFolderId ?? $this->_determineUploadFolderId($element, !$isDraftOrRevision); 
 };
```

In [Line 520](https://github.com/craftcms/cms/blob/3.5/src/fields/Assets.php#L520) you create a query to fetch all assets with `volume(:empty:)` but this does not work. It always fetches all assets. As an Example, this is your code
```PHP
$assetsToMove = Asset::find()
    ->id(ArrayHelper::getColumn($assets, 'id'))
    ->volumeId(':empty:')
    ->all();
```
And I changed it for debugging purpose to this.
```PHP
$query = Asset::find()
    ->id(ArrayHelper::getColumn($assets, 'id'))
    ->volumeId(':empty:');

echo $query->getRawSql();
$assetsToMove = $query->all();

echo "<pre>";
var_dump($assetsToMove[0]->id);
var_dump($assetsToMove[0]->volumeId);
var_dump($assetsToMove[0]->folderId);
echo "</pre>";
die();
```

The result is the following
> SELECT `elements`.`id`, `elements`.`fieldLayoutId`, `elements`.`uid`, `elements`.`enabled`, `elements`.`archived`, `elements`.`dateCreated`, `elements`.`dateUpdated`, `elements_sites`.`id` AS `elementSiteId`, `elements_sites`.`slug`, `elements_sites`.`siteId`, `elements_sites`.`uri`, `elements_sites`.`enabled` AS `enabledForSite`, `assets`.`volumeId`, `assets`.`folderId`, `assets`.`filename`, `assets`.`kind`, `assets`.`width`, `assets`.`height`, `assets`.`size`, `assets`.`focalPoint`, `assets`.`keptFile`, `assets`.`dateModified`, `volumeFolders`.`path` AS `folderPath`, `assets`.`uploaderId`, `content`.`id` AS `contentId`, `content`.`title`, `content`.`field_contactMessage`, `content`.`field_countPostsPage`, `content`.`field_countPostsModule`, `content`.`field_pageSizeMap`, `content`.`field_boolSendMail`, `content`.`field_autor`, `content`.`field_postId`, `content`.`field_userFullName`, `content`.`field_contactSubject`, `content`.`field_imagePosition`, `content`.`field_caption`, `content`.`field_body`, `content`.`field_latitude`, `content`.`field_fileName`, `content`.`field_description`, `content`.`field_contactEmail`, `content`.`field_endDate`, `content`.`field_color`, `content`.`field_question`, `content`.`field_googleMapsPinLatOffset`, `content`.`field_googleMapsPinLongOffset`, `content`.`field_googleMapsPinOffsetRadius`, `content`.`field_googleMapsLat`, `content`.`field_googleMapsLong`, `content`.`field_googleMapsZoom`, `content`.`field_greeting`, `content`.`field_headline`, `content`.`field_headlineLeft`, `content`.`field_headlineRight`, `content`.`field_showInFooter`, `content`.`field_showInMediacenter`, `content`.`field_longitude`, `content`.`field_lightswitch`, `content`.`field_linkField`, `content`.`field_contactName`, `content`.`field_dontShowInMenu`, `content`.`field_positionKarriere`, `content`.`field_searchindexFollow`, `content`.`field_seoTitle`, `content`.`field_startDate`, `content`.`field_activeStatus`, `content`.`field_subline`, `content`.`field_contactTelephone`, `content`.`field_text`, `content`.`field_textOverview`, `content`.`field_textUnformatted`, `content`.`field_textLeft`, `content`.`field_textRight`, `content`.`field_videoData`, `content`.`field_userLocation`
FROM (SELECT `elements`.`id` AS `elementsId`, `elements_sites`.`id` AS `elementsSitesId`, `content`.`id` AS `contentId`
FROM `craft_elements` `elements`
INNER JOIN `craft_assets` `assets` ON `assets`.`id` = `elements`.`id`
INNER JOIN `craft_elements_sites` `elements_sites` ON `elements_sites`.`elementId` = `elements`.`id`
INNER JOIN `craft_content` `content` ON (`content`.`elementId` = `elements`.`id`) AND (`content`.`siteId` = `elements_sites`.`siteId`)
> 
> WHERE (`elements_sites`.`siteId`=1) AND 
(`elements`.`id`=3232) AND 
(`elements`.`archived`=FALSE) AND 
(`elements`.`dateDeleted` IS NULL) AND 
(`elements_sites`.`enabled`=TRUE) AND 
(`elements`.`draftId` IS NULL) AND 
(`elements`.`revisionId` IS NULL)
> 
> ORDER BY `elements`.`dateCreated` DESC) `subquery`
INNER JOIN `craft_assets` `assets` ON `assets`.`id` = `subquery`.`elementsId`
INNER JOIN `craft_volumefolders` `volumeFolders` ON `volumeFolders`.`id` = `assets`.`folderId`
INNER JOIN `craft_elements` `elements` ON `elements`.`id` = `subquery`.`elementsId`
INNER JOIN `craft_elements_sites` `elements_sites` ON `elements_sites`.`id` = `subquery`.`elementsSitesId`
INNER JOIN `craft_content` `content` ON `content`.`id` = `subquery`.`contentId`
ORDER BY `elements`.`dateCreated` DESC

> int(3227) // file ID
> string(1) "3" // volume ID -> not empty
> string(2) "31" // folderId

Thus it will move the file. It would be nice if you could fix that.. [You removed your parseParam from here](https://github.com/craftcms/cms/blob/3.5/src/elements/db/AssetQuery.php#L845) please re-add it or use a normal where condition

### Additional info

- Craft version: 

Craft Pro 3.5.0-beta.3

**Comment by andris-sevcenko** [comment]
Thanks for reporting that!
This is actually a duplicate of #6217 and has been fixed for 3.5, but hasn't been released yet.
